snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -164,8 +170,10 @@ def finalize_replica(self) -> str:
         logger.info('Finalized replica image %s', self.replica_meta["name"])
         return replica_image.tags[0]
 
-    def _generate_credentials(self) -> Credentials:
-        host = DOCKER_TARGET_CONTAINER if IS_IN_DOCKER else 'localhost'
+    def regenerate_credentials(self, host) -> Credentials:
```

Comment:
Why do we need to regenerate the credentials? Is there something that can't be moved to the `__init__` function?

Reply:
The `regenerate_credentials()` method is used to regenerate if a target image is passed  which might be different from the default target image